changelog.d/pr-7851

```diff
@@ -15,5 +15,7 @@ description: {
 - The name of the generated script executable has been changed from "script" to
   "cabal-script-<your-sanitized-script-name>" for easier process management.
 - Reduce the default verbosity of scripts, so that the build output doesn't interfere with the script output.
+- Scripts now support a project metadata block that allows them to use options
+  that would normally be set in a cabal.project file.
```

Comment:
>  this just reads it and does not attempt to validate that the fields set are relevant to scripts. I think this is probably fine, but I'm open to discussing it.

a note in docs about that would be enough imo

Reply:
And an issue about adding such validation would be great too

Reply:
I added an issue: #8024

Reply:
Many thanks, a really good recap, maybe it worths to link the issue in the doc note (as list all the problematic fields would be too verbose)